### PR TITLE
#175494114 Admin should assign users to their managers

### PR DESCRIPTION
**Pull request by LouangeMu**
#### What does this PR do?

- a user should be able to have a line manager
- allow admin to assign line manager

#### Description of Task to be completed?
Every user has a line manager who is in charge of the user(s) assigned to him,A user should be able to have a line manager So that, he/she can have a person in charge of all my trip requests
#### How should this be manually tested?

- open your terminal and clone this repo by running git clone `https://github.com/Soma-Technologies-Inc/visitAfrica-backend`
- run git checkout `ft-assign-line-manager-175494114`
- run `git pull` to have the updated code
- run `npm install` to install all needed packages
- run `npm run migrate` to create migrations for the database
- run `npm startDev`
- navigate to login route POST /api/auth/login and login to get token that will be used as an Authorization header for the superAdmin 
- use these body data { "email": "loua@gmail.com", "password": "GoodG@12" }
- you will get a jwt 
- Use this route in postman `POST /api/users/user/userId `and in the body you will put the ManagerId only
- before assigning the user to their manager make sure you are using the correct token of a superAdmin you got while loging in the superAdmin otherwise you won;t be able to accomplish the task .

- finally run `npm run test`

#### Any background context you want to provide?
N/A
#### Screenshots (if appropriate)

- The person who is not an Admin won't be able to assign users to their Manager

image

- The manager is the one who is capable of being assigned a user

image

- When the user is not in the Database he won't be assigned to a manager

image

- When the user and the manager accomplish the requirements 

image

### CODE REVIEW

**Comment by BenieSereine** [comment]
sound good to me

File: src/database/migrations/20201129004549-create-user.js [hunk 1]
```diff
@@ -51,6 +51,10 @@ export async function up(queryInterface, Sequelize) {
 			type: Sequelize.STRING,
 			defaultValue: 'Requester',
 		},
+		lineManager: {
```
Comment (Niyonsengaeric) [inline src/database/migrations/20201129004549-create-user.js]: we don't update database like that you must create another migration to update the existing one!

File: src/database/seeders/20201215130557-trips.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+module.exports = {
```
Comment (Niyonsengaeric) [inline src/database/seeders/20201215130557-trips.js]: why are you creating this table ?

File: src/middlewares/user.middleware.js [hunk 3]
```diff
@@ -39,7 +39,6 @@ export const checkUserCredentials = async (req, res, next) => {
 };
 export const checkUserIfExist = async (req, res, next) => {
```
Comment (Niyonsengaeric) [inline src/middlewares/user.middleware.js:40]: change this name to the one corresponding to the task that is being done here

File: src/routes/user.route.js [hunk 4]
```diff
@@ -2,12 +2,15 @@ import { Router } from 'express';
 import UserController from '../controllers/user.controller';
 import protectRoute from '../middlewares/protect-route.middleware';
 import { checkUserIfExist } from '../middlewares/user.middleware';
-import { validateProfilePage, validateUserRoleBody } from '../validations/user.validation';
+import { validateProfilePage, validateUserRoleBody, validateUserBody } from '../validations/user.validation';
+import verifySuperAdmin from '../middlewares/superAdmin.verify.middleware';
+import userManagement from '../middlewares/user_manager.middleware';
```
Comment (Niyonsengaeric) [inline src/routes/user.route.js]: put the meaningful name

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

**Comment by tigthor** [comment]
@LouangeMu  Work on your pr description and how this can be manually tested

File: src/services/user.service.js [hunk 5]
```diff
@@ -51,6 +51,17 @@ class UserService {
 		return User.findOne({ where: property });
 	}
 
+	/**
+   * @param {integer} managerId id of the manager to be finding
+   * @returns {object} data of the manager found
+	*/
+	static async assignUserManager(managerId) {
+		const managerData = await Querie.assignUserManager(db.user, managerId);
```
Comment (tigthor) [inline src/services/user.service.js]: Why are you using Querie.assig..... Please only use User model don't add Que.....

**Review by tigthor — CHANGES_REQUESTED** [review]
Work on the my comments.

File: src/middlewares/user.middleware.js (hunk 3 above)
Comment (LouangeMu) [reply src/middlewares/user.middleware.js:40]: This was for Hyacente .for me I verified the user ,I did not check

File: src/database/migrations/20201129004549-create-user.js (hunk 1 above)
Comment (LouangeMu) [reply src/database/migrations/20201129004549-create-user.js]: Resolved

File: src/database/seeders/20201215130557-trips.js (hunk 2 above)
Comment (LouangeMu) [reply src/database/seeders/20201215130557-trips.js]: Resolved

File: src/services/user.service.js (hunk 5 above)
Comment (LouangeMu) [reply src/services/user.service.js]: Changed

**Pull request merged**